cli_tools/common/utils/daisyutils/daisy_worker.go

```diff
@@ -117,18 +179,22 @@ func (w *defaultDaisyWorker) RunAndReadSerialValue(key string, vars map[string]s
 func (w *defaultDaisyWorker) RunAndReadSerialValues(vars map[string]string, keys ...string) (map[string]string, error) {
 	err := w.Run(vars)
 	m := map[string]string{}
-	for _, key := range keys {
-		m[key] = w.wf.GetSerialConsoleOutputValue(key)
+	if w.finished != nil {
+		for _, key := range keys {
+			m[key] = w.finished.GetSerialConsoleOutputValue(key)
+		}
 	}
 	return m, err
 }
 
 func (w *defaultDaisyWorker) Cancel(reason string) bool {
-	if w.wf != nil {
-		w.wf.CancelWithReason(reason)
-		return true
-	}
-
-	//indicate cancel was not performed
-	return false
+	// do-once is required to ensure that additional calls
```

Comment:
What is `do-once`? `Do` is called below.

Reply:
Updated to `once.Do`  (`do-once` is just what made sense in my head, but clearly wasn't that great :)  )